src/Gzdecode.php

```diff
@@ -139,10 +139,12 @@ class Gzdecode
      *
      * @param string $name
      * @param mixed $value
+     *
+     * @throws \Exception
      */
     public function __set(string $name, $value)
```

Comment:
I'd be fine with that, but kept my changes to the minimum needed to fix the PHP 8.4 deprecations. Removing the method is a change which feels to me like something which should be decided by maintainer, not by a drive-by contributor like me.

Reply:
On second thought, in this case, it's better to keep the method as it is a more stringent safeguard against dynamic properties than relying on users of this package to be running static analysis tools.

Reply:
> On second thought, in this case, it's better to keep the method as it is a more stringent safeguard against dynamic properties than relying on users of this package to be running static analysis tools.

I doubt anyone sets any properties on this, they are mostly internal state only read by the parse method and initialized at the start.

We could just as well mark them private since they are already `@access private` but that’s also for another PR.

Reply:
Indeed, but that still won't protect against dynamic properties (even making a class `final` won't).
Would be interesting to see what the reason was the method got added. I imagine it could be something to do with security and/or serialization ?

Reply:
The class is just a pure PHP implementation of [`gzdecode`](https://www.php.net/manual/en/function.gzdecode.php) function (which was only introduced with PHP 5.4). The class is used to encapsulate the state of the stateful `gzip` decompression algorithm.

My best guess is Sam got bitten by a typo so she added it to avoid confusing debugging in the future https://github.com/simplepie/simplepie/commit/0aa49e5fbe77bd04838bf1bab7c794c095c2d5ba

We should probably deprecate it in favour of `gzdecode` too.

Reply:
Sounds like a good idea.